src/firefly/js/drawingLayers/WebGrid.js

```diff
@@ -0,0 +1,882 @@
+                || csys===CoordinateSys.EQ_B2000
+                || csys===CoordinateSys.EQ_J2000 ) {
+                //assign the new lower and upper longitude if found
+                if (vals[0] < range[0][0]) range[0][0] = vals[0];
+                if (vals[0] > range[0][1]) range[0][1] = vals[0];
+
+                //assign the new lower and upper latitude if found
+                if (vals[1] < range[1][0]) range[1][0] = vals[1];
+                if (vals[1] > range[1][1]) range[1][1] = vals[1];
+            }
+        }
+        x += dx;
+        y += dy;
+
+        ++i;
+    }
+
+}
+ /**
+  * get the ege value
+  * @param intervals
+  * @param width
+  * @param height
+  * @param csys
+  * @param wrap
+  * @param cc
+  * @returns {*[]}
+  */
+ function  edgeVals(intervals, width, height, csys, wrap, cc) {
+
+    var  range = [[1.e20, -1.e20],[1.e20, -1.e20]];
+    var xmin=0;
+    var ymin=0;
+    var xmax= width;
+    var ymax= height;
+    var xdelta, ydelta, x, y;
+
+
+    // four corners.
+    // point a[xmin, ymax], the top left point, from here toward to right top point b[xmax, ymax], ie the line
+    //   a-b
+    y = ymax;
+    x = xmin;
+    xdelta = (width / intervals) - 1; //define an interval of the point in line a-b
+    ydelta = 0; //no change in the y direction, so ydelta is 0, thus the points should be alone line a-b
+    edgeRun(intervals, x, y, xdelta, ydelta, range, csys,wrap, cc);
+
+    // Bottom: right to left
+    y = ymin;
+    x = xmax;
+    xdelta = -xdelta;
+    edgeRun(intervals, x, y, xdelta, ydelta, range, csys, wrap, cc);
+
+    // Left.  Bottom to top.
+    xdelta = 0;
+    ydelta = (height / intervals) - 1;
+    y = ymin;
+    x = xmin;
+    edgeRun(intervals, x, y, xdelta, ydelta, range, csys, wrap, cc);
+
+    // Right. Top to bottom.
+    ydelta = -ydelta;
+    y = ymax;
+    x = xmax;
+    edgeRun(intervals, x, y, xdelta, ydelta, range, csys, wrap,cc);
+
+    // grid in the middle
+    xdelta = (width / intervals) - 1;
+    ydelta = (height / intervals) - 1;
+    x = xmin;
+    y = ymin;
+    edgeRun(intervals, x, y, xdelta, ydelta, range, csys, wrap,cc);
+
+
+    return range;
+}
+ /**
+  * Test to see if the edge is a real one
+  * @param xrange
+  * @param trange
+  * @returns {boolean}
+  */
+ function  testEdge( xrange, trange)
+{
+    /* This routine checks if the experimental minima and maxima
+     * are significantly changed from the old test minima and
+     * maxima.  xrange and trange are assumed to be multidimensional
+     * extrema of the form double[ndim][2] with the minimum
+     * in the first element and the maximum in the second.
+     *
+     * Note that xrange is modified to have the most extreme
+     * value of the test or old set of data.
+     */
+
+    var delta = [];
+
+    /* Find the differences between the old data */
+    for (let i=0; i<trange.length; i+=1) {
+        delta[i] = Math.abs(trange[i][1]-trange[i][0]);
```

Comment:
you might be able to use `map` not quite sure. Also if the array is really long then it might be too inefficient.